xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -232,6 +239,80 @@ public FeeResult fee() throws JsonRpcClientErrorException {
     return jsonRpcClient.send(request, FeeResult.class);
   }
 
+  /**
+   * Get the ledger index of a tx result response. If not present, throw an exception.
+   *
+   * @return A string containing value of last validated ledger index.
+   */
+  private UnsignedInteger getMostRecentlyValidatedLedgerIndex() throws JsonRpcClientErrorException {
+    JsonRpcRequest request = JsonRpcRequest.builder()
+      .method(XrplMethods.LEDGER)
+      .addParams(LedgerRequestParams.builder().ledgerSpecifier(LedgerSpecifier.VALIDATED).build())
+      .build();
+    return jsonRpcClient.send(request, LedgerResult.class).ledgerIndexSafe().unsignedIntegerValue();
+  }
+
+  /**
+   * Check if the transaction is final on the ledger or not.
+   *
+   * @param transactionHash {@link Hash256} of the submitted transaction to check the status for.
+   * @param submittedOnLedgerIndex {@link LedgerIndex} on which the transaction with hash transactionHash
+   *        was submitted. This can be obtained from submit() response of the tx as validatedLedgerIndex.
+   * @return {@code true} if the {@link Transaction} is final/validated else {@code false}.
+   * @throws JsonRpcClientErrorException if {@code jsonRpcClient} throws an error.
+   */
+  public boolean isFinal(Hash256 transactionHash, LedgerIndex submittedOnLedgerIndex)
+    throws JsonRpcClientErrorException {
+    JsonRpcRequest request = JsonRpcRequest.builder()
+      .method(XrplMethods.TX)
+      .addParams(TransactionRequestParams.of(transactionHash))
+      .build();
+
+    TransactionResult response = jsonRpcClient.send(request, TransactionResult.class);
+    UnsignedInteger lastLedgerSequence = response.transaction().lastLedgerSequence()
```

Comment:
What happens here if the transaction has already expired (ie lastLedgerSequence has already passed)? I believe your call to `this.transaction()` will throw an exception because you won't be able to find the transaction on ledger. 

This ends up being sort of paradoxical -- You can't check to see if a transaction is expired without checking the `lastLedgerIndex`, but you can't get `lastLedgerIndex` without getting the transaction from the ledger, but if the condition you are looking for is true, you will never get the transaction from ledger, and therefore can never get `lastLedgerSequence`.

The solution to this is that users have to record the `lastLedgerSequence` that they set on the transaction _before_ submitting it, and you need to allow them to pass that `lastLedgerSequence` into this method.